report-generator/src/main/java/com/teamscale/report/testwise/jacoco/JaCoCoTestwiseReportGenerator.java

```diff
@@ -47,15 +47,14 @@
 	public JaCoCoTestwiseReportGenerator(Collection<File> codeDirectoriesOrArchives,
 										 ClasspathWildcardIncludeFilter locationIncludeFilter,
 										 EDuplicateClassFileBehavior duplicateClassFileBehavior,
-										 ILogger logger) throws CoverageGenerationException {
+										 ILogger logger) {
 		this.locationIncludeFilter = locationIncludeFilter;
-		this.executionDataReader = new CachingExecutionDataReader(logger);
-		this.executionDataReader
-				.analyzeClassDirs(codeDirectoriesOrArchives, locationIncludeFilter, duplicateClassFileBehavior);
```

Comment:
I might be blind, but I did not see where `analyzeClassDirs` is called now

Reply:
whoops, I seem to have undone this change by accident :P

Reply:
I moved it into the execution data reader :-) it's transparent now for the caller and happens when the first dump is performed.